src/traveler.js

```diff
@@ -1,12 +1,14 @@
+import User from './user';
 import Trip from './trip';
 
-class Traveler {
-  constructor(travelerInfo, myTrips) {
+class Traveler extends User {
+  constructor(userType, myTrips, travelerInfo) {
+    super(userType, myTrips);
     this.id = travelerInfo.id;
     this.name = travelerInfo.name;
     this.travelerType = travelerInfo.travelerType;
     this.myTrips = myTrips;
-    this.totalSpent = this.getTotalAmountSpentThisYear();
+    this.totalSpent = this.getTotalAmountSpentThisYear(myTrips, userType);
```

Comment:
This is a good case use for Object Inheritance.  I noticed that you created different children classes from the User class.  This is a good way to dry the overall code in a logical pattern.